docs/workflows/copy.rst

```diff
@@ -1,43 +1,189 @@
+copy content in the background which you, the user, did not explicitly tell it to copy. For
+example:
+
+* When copying an Advisory (Errata) from one repository to another, all of the RPM packages
+  directly referenced by the Advisory will also be copied.
+* When copying a Module from one repository to another, all of its modular artifacts
+  (RPM packages) will also be copied.
+
+The goal is to be as easy to use as possible while maintaining the "correctness" of the repository.
+
+Dependency solving
+__________________
+
+When copying RPM packages between repositories, it is likely desirable to ensure that all of the
+RPM dependencies they need to be "installable" on a client system are also present in the
+repository they are being copied to. For example, if you want to add the "hexchat" RPM to a
+new repository, and you want to be able to install it from that repository, the repository should
+also contain its dependencies such as "libnotify" and "gtk3". This applies likewise to all RPM
+concepts that "contain" RPMs also, such as Advisories (Errata), Modules, and PackageGroups. With
+the RPM copy API, you are afforded the option to have all dependencies (and the dependencies of
+those dependencies) copied for you automatically in the background, if they do not already exist
+in the destination repository. In Pulp 2, this feature was called "recursive copy" because in a
+sense it copies dependencies recursively.
+
+Unfortunately, solving these complex dependency relationships can be quite expensive. For this
+reason, the functionality to perform these checks is not enabled by default, and is instead
+provided by setting the "dependency_solving" parameter to a value of True when making calls against
+the API. Note that if you do elect to not perform dependency solving, it will be possible to
+accidentally create incomplete repositories, if you are not careful.
+
+.. note::
+
+    While the default value for this "dependency_solving" parameter is currently "false", this
+    default is potentially subject to change in the future - until such a time as this API is
+    stabilized.
+
+Dependency solving does have some restrictions to be aware of. The set of content contained by
+all repositories used in a copy operation must be "dependency closed", which is to say that no
+content in any repository may have a dependency which cannot be satisfied by any content present
+in any of the other repositories involved in the copy operation.
+
+For example, in CentOS 8, there are two primary repositories which are called "BaseOS" and
+"AppStream". RPMs present in the "AppStream" repository frequently depend on RPMs which are
+not present in "AppStream", but are present in "BaseOS", instead.
+
+In order to copy RPMs from a Pulp-clone of the "AppStream" repository, you must perform a
+"multi-repository copy" so that the dependencies can be properly resolved. Please see the
+recipe section below for more details on how to do this.
```

Comment:
Should it have it's own heading? Like "Restrictions" or "Requirements". Up to you, just a suggestion to draw more attention to this section.